src/debugger/launch.ts

```diff
@@ -35,6 +40,29 @@ export async function makeDebugConfigurations(ctx: FolderContext) {
     for (const config of configs) {
         const index = launchConfigs.findIndex(c => c.name === config.name);
         if (index !== -1) {
+            if (keysToUpdate && launchConfigs[index] !== config) {
```

Comment:
Can we break this update code into a separate function in utilities.ts. It seems quite useful

Reply:
Oh plus maybe add some comments detailing what it is doing

Reply:
🤔This doesn’t appear as easy as it looks like… We’re dancing at the edge of TypeScript’s type system here, and I tried many possible generics but still cannot comfort the type checker.

Reply:
This means we can only specify single type (`vscode.DebugConfiguraion`) to use it, which I think is not so general to go into `utilities`. Just extract to a fileprivate function in `launch.ts` instead?

Reply:
Sounds good